packages/commerce-sdk-react/src/hooks/useAuthenticatedRequest.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {QueryFunctionContext, QueryKey} from '@tanstack/react-query'
+import useAuth from './useAuth'
+import useCommerceApi from './useCommerceApi'
+import {ApiClients, QueryFunction, MutationFunction} from './types'
+
+interface Client {
+    clientConfig: {
+        headers: Record<string, string>
+    }
+}
+
+function useAuthenticatedMutation<TData, TVariables = unknown>(
+    fn: MutationFunction<TData, TVariables>
+) {
+    const auth = useAuth()
+    const apiClients = (useCommerceApi() as unknown) as Record<string, Client>
+
+    return (variables: TVariables) => {
+        return auth
+            .ready()
+            .then(({access_token}) => {
+                Object.keys(apiClients).forEach((client) => {
+                    apiClients[client].clientConfig.headers = {
+                        ...apiClients[client].clientConfig.headers,
+                        Authorization: `Bearer ${access_token}`
+                    }
+                })
+                return (apiClients as unknown) as ApiClients
```

Comment:
This probably old, why `apiClients` is deferred as unknown and then ApiClients here?  🤔

Reply:
Because typescript requires you to cast it as unknown before anything else.

<img width="1038" alt="Screen Shot 2022-10-05 at 5 44 18 PM" src="https://user-images.githubusercontent.com/10948652/194188956-f8e18881-092a-427e-bc10-58f082091851.png">

In most cases, it's unsafe to do a type cast like this. But for this particular case, we know that all api clients have the same properties and we can cast it to a type so we can treat them as same type for loops.

Reply:
This issue has been fixed, the error happened because Typescript does not understand the values inside the array returning by Object.keys.